q2_gneiss/plot/_plot.py

```diff
@@ -49,56 +55,128 @@ def f(x):
         num_features = pd.DataFrame(
             {num_clade.name: taxa_df.loc[num_clade.name]}
             ).T
+        r = 1
     else:
         num_features = taxa_df.loc[num_clade.subset()]
+        r = len(list(num_clade.tips()))
 
     if denom_clade.is_tip():
         denom_features = pd.DataFrame(
             {denom_clade.name: taxa_df.loc[denom_clade.name]}
             ).T
+        s = 1
     else:
         denom_features = taxa_df.loc[denom_clade.subset()]
+        s = len(list(denom_clade.tips()))
+
+    balances = (np.log(table.loc[:, num_features.index]).mean(axis=1) -
+                np.log(table.loc[:, denom_features.index]).mean(axis=1))
+
+    balances = balances * np.sqrt(r * s / (r + s))
+    balances = pd.DataFrame(balances, index=table.index,
+                            columns=[balance_name])
 
-    num_color, denom_color = '#4c72b0', '#4c72b0'
+    # the actual colors for the numerator and denominator
+    num_color = '#105d33'
+    denom_color = '#b0d78e'
 
     fig, (ax_num, ax_denom) = plt.subplots(2)
     balance_barplots(tree, balance_name, taxa_level, taxa_df,
                      denom_color=denom_color, num_color=num_color,
                      axes=(ax_num, ax_denom))
 
     ax_num.set_title(
-        r'$%s_{numerator} \; taxa \; (%d \; taxa)$' % (balance_name,
-                                                       len(num_features)))
+        r'$%s_{numerator} \; taxa \; (%d \; taxa)$' % (
+            balance_name, len(num_features)))
     ax_denom.set_title(
-        r'$%s_{denominator} \; taxa \; (%d \; taxa)$' % (balance_name,
-                                                         len(denom_features)))
+        r'$%s_{denominator} \; taxa \; (%d \; taxa)$' % (
+            balance_name, len(denom_features)))
     ax_denom.set_xlabel('Number of unique taxa')
     plt.tight_layout()
     fig.savefig(os.path.join(output_dir, 'barplots.svg'))
     fig.savefig(os.path.join(output_dir, 'barplots.pdf'))
 
+    dcat = None
     if metadata is not None:
         fig2, ax = plt.subplots()
         c = metadata.to_series()
         data, c = match(balances, c)
         data[c.name] = c
         y = data[balance_name]
+
         # check if continuous
         try:
             c = c.astype(np.float64)
             ax.scatter(c.values, y)
             ax.set_xlabel(c.name)
+            if threshold is None:
+                threshold = c.mean()
+            dcat = c.apply(
+                lambda x: '%s < %f' % (c.name, threshold)
+                if x < threshold
+                else '%s > %f' % (c.name, threshold)
+            )
+
         except:
```

Comment:
Not strictly related, but since you are here, could you change this to `except Exception:`?

Reply:
Done!